packages/core/src/provider.tsx

```diff
@@ -63,92 +35,48 @@ function reducer(state: any, { type, payload }: any) {
 }
 
 export function AElfReactProvider({ children, appName, nodes: providerNodes }: AElfReactProviderProps) {
-  const [state, dispatch]: [AElfContextType, any] = useReducer(reducer, INITIAL_STATE);
+  const [state, dispatch] = useReducer(reducer, INITIAL_STATE);
+  const { account, defaultAElfBridge } = state;
   const activate = useCallback(
     async (activateNodes?: AElfReactProviderProps['nodes']) => {
       const nodes = activateNodes || providerNodes;
       if (!nodes) throw Error('activate must exist nodes');
-      // eslint-disable-next-line no-async-promise-executor
-      return new Promise(async (resolve, reject) => {
-        // @ts-ignore
-        const isAElfBridge = isMobileDevices() && !window?.NightElf;
-        const connector = (
-          await (isAElfBridge ? import('./utils/NightElf/AelfBridgeCheck') : import('./utils/NightElf/NightElfCheck'))
-        ).default;
-        connector
-          .getInstance()
-          .check.then(() => {
-            let firstKey = '';
-            const bridges: { [key: string]: AElfDappBridge } = {};
-            Object.entries(nodes).forEach(([k, v]) => {
-              if (!firstKey) firstKey = k;
-              bridges[k] = connector.initAelfInstanceByExtension(v.rpcUrl, appName);
-            });
-            const node = nodes[firstKey];
-            const bridge = bridges[firstKey];
-            bridge
-              .login({
-                chainId: node.chainId,
-                payload: {
-                  method: 'LOGIN',
-                },
-              })
-              .then(async (result) => {
-                if (result.error) {
-                  reject(result);
-                } else {
-                  if (isMobileDevices()) {
-                    await bridge.connect();
-                  } else {
-                    await Promise.all(Object.values(bridges).map((i) => i.chain.getChainStatus()));
-                  }
-                  const detail = formatLoginInfo(result.detail);
-                  dispatch({
-                    type: ACTIVATE,
-                    payload: { ...detail, defaultAElfBridge: bridge, aelfBridges: bridges, isActive: true, nodes },
-                  });
-                  resolve(true);
-                }
-              })
-              .catch((error) => {
-                reject(error);
-              });
-          })
-          .catch((error: any) => {
-            reject(error);
-          });
+      const { bridge, bridges, node } = await getBridges(nodes, appName);
+      const result = await bridge.login({ chainId: node.chainId, payload: { method: 'LOGIN' } }).then();
+      // login fail
+      if (result.error) throw result;
+
+      if (isMobileDevices()) {
```

Comment:
`// is Wallet APP or  an implementation of aelf-bridge` or `// is an implementation of NightElf`?